YoutubeDownloader/App.xaml

```diff
@@ -1,473 +1,109 @@
-                <ControlTemplate.Triggers>
-                    <Trigger Property="TickPlacement" Value="TopLeft">
-                        <Setter TargetName="TopTick" Property="Visibility" Value="Visible" />
-                    </Trigger>
-                    <Trigger Property="TickPlacement" Value="BottomRight">
-                        <Setter TargetName="BottomTick" Property="Visibility" Value="Visible" />
-                    </Trigger>
-                    <Trigger Property="TickPlacement" Value="Both">
-                        <Setter TargetName="TopTick" Property="Visibility" Value="Visible" />
-                        <Setter TargetName="BottomTick" Property="Visibility" Value="Visible" />
-                    </Trigger>
-                    <Trigger Property="IsSelectionRangeEnabled" Value="true">
-                        <Setter TargetName="PART_SelectionRange" Property="Visibility" Value="Visible" />
-                    </Trigger>
-                    <Trigger Property="IsEnabled" Value="False">
-                        <Setter TargetName="IncreaseRepeatButton" Property="Background" Value="{DynamicResource MaterialDesignCheckBoxDisabled}" />
-                    </Trigger>
-                </ControlTemplate.Triggers>
-            </ControlTemplate>
+    </Application.Styles>
 
-            <Style x:Key="MaterialDesignThinSlider" TargetType="{x:Type Slider}">
-                <Setter Property="Stylus.IsPressAndHoldEnabled" Value="false" />
-                <Setter Property="Background" Value="{x:Null}" />
-                <Setter Property="BorderBrush" Value="Transparent" />
-                <Setter Property="Foreground" Value="{DynamicResource PrimaryHueMidBrush}" />
-                <Setter Property="Template" Value="{StaticResource MaterialDesignSliderHorizontal}" />
-                <Style.Triggers>
-                    <Trigger Property="Orientation" Value="Vertical">
-                        <Setter Property="Template" Value="{StaticResource MaterialDesignSliderVertical}" />
-                    </Trigger>
-                    <Trigger Property="IsEnabled" Value="False">
-                        <Setter Property="Foreground" Value="{DynamicResource MaterialDesignCheckBoxDisabled}" />
-                    </Trigger>
-                </Style.Triggers>
-            </Style>
-        </s:ApplicationLoader>
+    <Application.Resources>
+        <Color x:Key="SuccessColor">#006400</Color>
+        <Color x:Key="CanceledColor">#FF8C00</Color>
+        <Color x:Key="FailedColor">#8B0000</Color>
+
+        <SolidColorBrush x:Key="SuccessBrush" Color="{DynamicResource SuccessColor}" />
+        <SolidColorBrush x:Key="CanceledBrush" Color="{DynamicResource CanceledColor}" />
+        <SolidColorBrush x:Key="FailedBrush" Color="{DynamicResource FailedColor}" />
+    
+        <!-- ControlThemes-->
+        <ControlTheme x:Key="MaterialDesignRoot" TargetType="{x:Type Control}" BasedOn="{StaticResource MaterialUserControl}">
```

Comment:
Question (as someone not familiar with Avalonia): what is the difference between a `Style` and ` ControlTheme`?

Reply:
ControlTheme is the equivalent to Styles in WPF. Styles is a Avalonia specific thing that is the default for doing some control styling. ControlTheme is target more for the Theme of a control. Styles have the advantage of being able to use several on a control. ControlThemes in Avalonia do not have an equivalent for StyleTriggers but Avalonia-Styles can have pseudo selectors like in web. also they can be applied conditionally.
Some usful links:
https://docs.avaloniaui.net/docs/get-started/wpf/styling (the whole coming form WPF guide is really useful)
https://docs.avaloniaui.net/docs/basics/user-interface/styling/
https://docs.avaloniaui.net/docs/reference/styles/pseudo-classes
https://docs.avaloniaui.net/docs/reference/styles/style-selector-syntax